tools/misc/mapping_expander.py

```
@@ -0,0 +1,212 @@
+
+        # Create a new row for each match found
+        for full_node_id in matches:
+            new_rows.append({
+                "source_node_id": full_node_id if reverse else fixed_value,
+                "target_node_id": fixed_value if reverse else full_node_id
+            })
+
+    return pd.DataFrame(new_rows), warnings_list, exceptions_list
+
+
+def run_expansion(reverse=False):
```

Pylint (3.3.7)

[refactor] Too many local variables (22/15)

(R0914)